contrib/bindings/go/pathrs/pathrs.go

```diff
@@ -0,0 +1,328 @@
+// Copyright (C) 2019, 2020 Aleksa Sarai <cyphar@cyphar.com>
+// Copyright (C) 2020 Maxim Zhiburt <zhiburt@gmail.com>
+// Copyright (C) 2019, 2020 SUSE LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package pathrs provides a bindings for libpathrs, a library for safe path resolution on Linux.
+package pathrs
+
+// #cgo LDFLAGS: -lpathrs
+// #include <pathrs.h>
+import "C"
+import (
+	"fmt"
+	"math/rand"
+	"os"
+	"strings"
+	"syscall"
+	"time"
+	"unsafe"
+)
+
+// Root holds the responsibility to provide safe api to functions of pathrs root api
+type Root struct {
+	root *C.pathrs_root_t
+}
+
+// Handle represents an handle pathrs api interface
+type Handle struct {
+	handle  *C.pathrs_handle_t
+	nameGen *rand.Rand
+}
+
+// Open opens directory as a root directory
+func Open(path string) (*Root, error) {
+	root := C.pathrs_open(C.CString(path))
+	err := handleErr(C.PATHRS_ROOT, unsafe.Pointer(root))
+	if err != nil {
+		return nil, err
+	}
+
+	return &Root{root: root}, nil
+}
+
+// RootFromFd constructs a new file-based libpathrs object of root from a file descriptor
+// Uses often in combination with Root.IntoFd methood
+func RootFromFd(fd int) (*Root, error) {
+	root := (*C.pathrs_root_t)(C.pathrs_from_fd(C.PATHRS_ROOT, C.int(fd)))
+	err := handleErr(C.PATHRS_ROOT, unsafe.Pointer(root))
+	if err != nil {
+		return nil, err
+	}
+
+	return &Root{root: root}, nil
+}
+
+// HandleFromFd constructs a new file-based libpathrs object of handle from a file descriptor
+// Uses often in combination with Handle.IntoFd methood
+func HandleFromFd(fd int) (*Handle, error) {
```

Comment:
Should be moved along with the `struct Handle` definition. Honestly I actually missed this the first time I read through the change.